Add STAC vs ODC section to odc.stac docs (#410)
opendatacube/odc-tools: libs/stac/docs/stac-vs-odc.rst

STAC vs Open Datacube
#####################

> `Open Datacube`_ project on which this library is based started before `STAC`_
> spec existed. As a result ODC uses different terminology for otherwise very
> similar concepts.

Reviewer: ```suggestion
The `Open Datacube`_ (ODC) project, on which this library is based, started before `STAC`_
spec existed. As a result ODC uses different terminology for otherwise very
similar concepts.
```

.. list-table::
   :header-rows: 1

   * - STAC
     - ODC
     - Description
   * - :class:`~pystac.Collection`
     - Product or :class:`~datacube.model.DatasetType`
     - Collection of observations across space and time
   * - :class:`~pystac.Item`
     - :class:`~datacube.model.Dataset`
     - Single observation (specific time and place), multi-channel
   * - :class:`~pystac.Asset`
     - :class:`~datacube.model.Measurement`
     - Component of a single observation

>    * - Band

Reviewer: Maybe worth linking to https://github.com/stac-extensions/eo#band-object?

     - :class:`~datacube.model.Measurement`
     - Pixel plane within a multi-plane asset
   * - Common Name
     - Alias
     - Refer to the same band by different names

Similarly to STAC, ODC uses several levels of hierarchy to model metadata. At
the highest level there is *Product* which is a collection of *Datasets*. Each
*Dataset* contains a set of *Measurements* and related metadata. Finally
*Measurement* describes a single plane of pixels captured at roughly the same
time. Metadata includes location of the "file" and possibly location within a
file.

Multiple Bands per File
=======================

Multiple bands in a single file are supported by both ODC and STAC, but

> representation differs. In STAC another level of hierarchy is added below an
> *Asset*. Resource pointed by an *Asset* may contain more than one band of
> pixels, and *Asset* contains description of those bands. In ODC, *Asset* is not

Reviewer: ```suggestion
representation differs. In STAC another level of hierarchy is added below an
*Asset* via the [bands attribute of the EO extension](https://github.com/stac-extensions/eo#band-object). Resources pointed to by an *Asset* may contain more than one band of
pixels, and an *Asset* contains descriptions of those bands. In ODC, *Asset* is not
```

modelled explicitly, instead resource path and potential location within this

> resource is just a property of a *Measurement* object. It is common in STAC to

Reviewer: ```suggestion
resource are properties of a *Measurement* object. It is common in STAC to
```

> have one to one mapping between band and asset, in that scenario ODC

Reviewer: ```suggestion
have one to one mapping between band and asset, and in that scenario ODC
```

*Measurement* and STAC *Asset* can be seen as equivalent.

Geo Referencing Metadata
========================

> Precise geo referencing metadata is stored within a file pointed by

Reviewer: ```suggestion
Precise geo referencing metadata is stored within a file pointed to by
```

*Asset*/*Measurement*, but it can also be recorded within a STAC *Item*/ODC
*Dataset* document. Having geo-referencing information at this level can enable

> more efficient data access.

Reviewer: ```suggestion
more efficient data access by providing spatial information without needing to access the source (data file) itself.
```

> In STAC `Projection Extension`_ is used to bring this metadata from file to

Reviewer: ```suggestion
In STAC, the `Projection Extension`_ is used to bring this metadata from file to
```
